js/ui/codeView.js

```diff
@@ -1013,17 +1016,18 @@ var CodeViewManager = new Lang.Class({
         // This is a new proxy window, make it join the session
         if (proxy) {
             let variant = proxy.get_cached_property('Target');
-            let [targetBusName, targetObjectPath] = variant.deep_unpack();
+            let [targetAppId, targetWindowId] = variant.deep_unpack();
             let session = this._getSessionForTargetApp(
-                targetBusName, targetObjectPath);
+                targetAppId, targetWindowId);
 
             if (session)
                 handled = session.admitToolboxWindowActor(actor);
         } else {
+            let shellApp = Shell.WindowTracker.get_default().get_window_app(actor.meta_window);
```

Comment:
Are we guaranteed to have an app here? When I was working on Clubhouse Shell buttons I noticed that I couldn't get the app for the Clubhouse based on its actor's meta window (I didn't have time to investigate further, and using the GTK ID worked for me).

Reply:
Yeah, side components are treated specially in that respect for some reason, and would not be returned by that function, but it will work for any of the regular apps.